### PR TITLE
Alerting: Document state history config options in default and sample configuration files

### PR DESCRIPTION
**Pull request by alexweav**
**What is this feature?**

Initially, we used lack of documentation of these options as a pseudo-feature-flag.

We recently added proper feature toggles for alert state history. With that in place, we can now document the configuration options.

**Who is this feature for?**

Administrators, operators, developers

**Which issue(s) does this PR fix?**:

n/a

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.
- [x] There are no known compatibility issues with older supported versions of Grafana, or plugins.
- [x] It passes the [Hosted Grafana feature readiness review](https://docs.google.com/document/d/1QL9Ly8KnXzpb6ISbg49pTODRO5mhA5tkkfIZVX6pqQU/edit) for observability, scalability, performance, and security.

### CODE REVIEW

**Review by gotjosh — APPROVED** [review]
LGTM

File: conf/defaults.ini [hunk 1]
```diff
@@ -1010,6 +1010,54 @@ disabled_labels =
 # Enable the state history functionality in Unified Alerting. The previous states of alert rules will be visible in panels and in the UI.
 enabled = true
 
+# Choose which pluggable state history backend to use. Either "annotations", "loki", or "multiple"
+# "loki" will write state history to an external Loki instance. "multiple" allows history to be written to mulitple backends at once.
+# Defaults to "annotations".
```
Comment (brendamuir) [inline conf/defaults.ini:1015]:
```suggestion
# Select which pluggable state history backend to use. "annotations", "loki", or "multiple"
# "loki" writes state history to an external Loki instance. "multiple" allows history to be written to multiple backends at once.
# Defaults to "annotations".
```

File: conf/sample.ini [hunk 2]
```diff
@@ -967,6 +967,56 @@
...
 ;disabled_labels =
 
+[unified_alerting.state_history]
+# Enable the state history functionality in Unified Alerting. The previous states of alert rules will be visible in panels and in the UI.
+; enabled = true
+
+# Choose which pluggable state history backend to use. Either "annotations", "loki", or "multiple"
+# "loki" will write state history to an external Loki instance. "multiple" allows history to be written to mulitple backends at once.
+# Defaults to "annotations".
+; backend = "multiple"
```
Comment (brendamuir) [inline conf/sample.ini:977]: same as above

**Review by JohnnyQQQQ — APPROVED** [review]
LGTM

**Pull request merged**